impl/client/src/main/javascript/web/util/_a11y.js

```diff
@@ -35,6 +35,421 @@ define(["./_focus"], function(focusUtil) {
+          }
+        }
+      }
+      keysSoFar += character;
+      clearKeysSoFarAfterDelay();
+
+      var nextMatch = findMatchInRange(
+          itemList,
+          searchIndex + 1,
+          itemList.length
+      );
+      if (!nextMatch) {
+        nextMatch = findMatchInRange(itemList, 0, searchIndex);
+      }
+      return nextMatch;
+    }
+
+    function findMatchInRange(
+        list,
+        startIndex,
+        endIndex
+    ) {
+      // Find the first item starting with the keysSoFar substring, searching in
+      // the specified range of items
+      for (var n = startIndex; n < endIndex; n++) {
+        var label = list[n].innerText;
+        if (label && label.toUpperCase().indexOf(keysSoFar) === 0) {
+          return list[n];
+        }
+      }
+      return null;
+    }
+
+    /* Return the previous listbox option, if it exists; otherwise, returns null */
+    function findPreviousOption(currentOption) {
+      var allOptions = Array.prototype.slice.call(
+          listboxNode.querySelectorAll('[role="option"]')
+      ); // get options array
+      var currentOptionIndex = allOptions.indexOf(currentOption);
+      var previousOption = null;
+
+      if (currentOptionIndex > -1 && currentOptionIndex > 0) {
+        previousOption = allOptions[currentOptionIndex - 1];
+      }
+
+      return previousOption;
+    }
+
+    /* Return the next listbox option, if it exists; otherwise, returns null */
+    function findNextOption(currentOption) {
+      var allOptions = Array.prototype.slice.call(
+          listboxNode.querySelectorAll('[role="option"]')
+      ); // get options array
+      var currentOptionIndex = allOptions.indexOf(currentOption);
+      var nextOption = null;
+
+      if (currentOptionIndex > -1 && currentOptionIndex < allOptions.length - 1) {
+        nextOption = allOptions[currentOptionIndex + 1];
+      }
+
+      return nextOption;
+    }
+
+    /* Return the index of the passed element within the passed array, or null if not found */
+    function getElementIndex(option, options) {
+      var allOptions = Array.prototype.slice.call(options); // convert to array
+      var optionIndex = allOptions.indexOf(option);
+
+      return typeof optionIndex === 'number' ? optionIndex : null;
+    }
+
+    /**
+     * @description
+     *  Toggle the aria-selected value
+     * @param element
+     *  The element to select
+     */
+    function toggleSelectItem(element) {
+      if (multiselectable) {
+        var isAriaSelected = (element.getAttribute('aria-selected') === 'true');
+        if (isAriaSelected) {
+          element.setAttribute('aria-selected', 'false');
+          element.classList.remove('selected');
+        } else {
+          element.setAttribute('aria-selected', 'true');
+          element.classList.add('selected');
+        }
+
+      }
+    }
+
+    /**
+     * Check if the selected option is in view, and scroll if not
+     */
+    function updateScroll(key) {
```

Comment:
Only ever called with key up or down, so why not receive a boolean named, e.g. `alignTop` instead?

Reply:
updated with position relative on parent div and the key is removed.